test/test_nn.py

```diff
@@ -13371,6 +13353,76 @@ def test_grid_sample_nan_inf(self, device, dtype):
                                                      padding_mode=padding_mode, align_corners=False)
             self.assertEqual(sample, torch.zeros([1, 1, 1, 2], device=device, dtype=dtype))
 
+    @onlyOnCPUAndCUDA
+    def test_fractional_max_pool2d(self, device):
+        x = torch.randn(1, 2, 7, 7, requires_grad=True, device=device)
+        samples = x.new(1, 2, 2).uniform_()
+
+        def func(x):
+            return F.fractional_max_pool2d(
+                x, (2, 2), output_size=(3, 3), _random_samples=samples)
+
+        self.assertEqual(func(x).shape, (1, 2, 3, 3))
+        gradcheck(func, [x])
+        gradgradcheck(func, [x])
+
+        x = torch.randn(2, 7, 7, requires_grad=True, device=device)
+        self.assertEqual(func(x).shape, (2, 3, 3))
+        if self.device_type != 'cuda':
+            # Raises -> RuntimeError: TensorAccessor expected 4 dims but tensor has 3
+            # on CUDA in gradcheck
+            gradcheck(func, [x])
```

Comment:
In CUDA there is an error due to the following part of code

https://github.com/pytorch/pytorch/blob/bb9e0c625ebe0e8dd73d896b8a52ff807d969fb3/aten/src/ATen/native/cuda/FractionalMaxPool2d.cu#L266-L272

Reply:
Could you please open an issue saying that CUDA's fractional_maxpool does not support an input without batch dimension. For this PR, does it work if you add a size 1 batch dimension? Something like...

```
x = torch.randn(1, 2, 7, 7, requires_grad=True, device=device)
```

Reply:
The previous input has the batch dimension which works for both CUDA and CPU
This one does'nt on CUDA. Have opened an issue https://github.com/pytorch/pytorch/issues/52427 for the same and added the reference.